code/obj/submachine/seed.dm

```diff
@@ -826,18 +831,34 @@
 
 		..()
 
+	proc/remove_distant_beaker()
+		// borgs and people with item arms don't insert the beaker into the machine itself
+		// but whenever something would happen to the dispenser and the beaker is far it should disappear
+		if(src.inserted && BOUNDS_DIST(src.inserted, src) > 0)
+			if (src.inserted == src.extract_to) src.extract_to = null
+			src.inserted = null
+			src.UpdateIcon()
+
 	proc/tryInsert(var/obj/item/W, var/mob/user)
-		if (isrobot(user))
-			boutput(user, "This machine does not accept containers from robots!")
+		remove_distant_beaker()
+
+		if(BOUNDS_DIST(user, src) > 0) // prevent message from appearing in case a borg inserts from afar
 			return
+
 		if(src.inserted)
 			boutput(user, "<span class='alert'>A container is already loaded into the machine.</span>")
 			return
 		src.inserted =  W
-		user.drop_item()
-		W.set_loc(src)
-		if(!src.extract_to) src.extract_to = W
-		boutput(user, "<span class='notice'>You add [W] to the machine!</span>")
+
+		if(!W.cant_drop)
+			user.drop_item()
+			if(!QDELETED(W))
+				W.set_loc(src)
+		if(W.qdeled)
```

Comment:
```suggestion
		if(QDELETED(W))
```
I didn't make the same suggestion twice, there were just two instances of the same issue